casanovo/denovo/model.py

```diff
@@ -457,39 +457,44 @@ def on_test_epoch_end(self):
                 scores = batch[2].cpu()  # transfer to cpu in case in gpu
 
                 for i in range(len(batch[0])):
-                    top_scores = torch.max(scores[i], axis=1)[
-                        0
-                    ]  # take the score of most probable AA
-                    empty_index = torch.where(top_scores == 0.04)[
-                        0
-                    ]  # find the indices of positions after stop token
-
-                    if len(empty_index) > 0:  # check if decoding was stopped
-                        last_index = (
-                            empty_index[0] - 1
-                        )  # select index of the last AA
-
-                        if (
-                            last_index >= 1
-                        ):  # check if peptide is at least one AA long
-                            top_scores_list = top_scores[
-                                :last_index
-                            ].tolist()  # omit the stop token
+                    peptide_seq = batch[1][i][1:]
+                    # TODO: Compute the predicted mass from the peptide sequence.
```

Comment:
In case it's helpful, here's code to do this (copied from elsewhere):

```
AA_MASSES = {
    'A':71.03711,
    'C':103.00919,
    'D':115.02694,
    'E':129.04259,
    'F':147.06841,
    'G':57.02146,
    'H':137.05891,
    'I':113.08406,
    'J':113.08406,
    'K':128.09496,
    'L':113.08406,
    'M':131.04049,
    'N':114.04293,
    'O':114.07931,
    'P':97.05276,
    'Q':128.05858,
    'R':156.10111,
    'S':87.03203,
    'T':101.04768,
    'U':150.04344,
    'V':99.06841,
    'W':186.07931,
    'X':0, # N.B. Missing amino acid!
    'Y':163.06333,
}

MASS_WATER = 18.010564683

def compute_molecular_weight(sequence):
    """Compute the molecular weight of a protein."""
    global MASS_WATER, AA_MASSES
    
    return_value = MASS_WATER
    
    for letter in sequence:
        return_value += AA_MASSES[letter]

    return return_value
```

Reply:
Pyteomics and spectrum_utils have functions to do that as well, I just didn't get around to it yet yesterday.

The open question is where to get the precursor m/z from, but it should be possible. I'll have to try to understand the code a bit more.

Reply:
Depthcharge also has a calculator too!

Reply:
I'm using depthcharge to avoid adding extra dependencies, but I think it doesn't handle the modifications properly. It does have the MassIVE-KB mods in its mass dictionary, but [this line](https://github.com/wfondrie/depthcharge/blob/5246ef8b6414855918e0823836d2432c21db63ac/depthcharge/masses.py#L78) evaluates each character individually, rather then the modified residues as an atomic unit. Can you have a look at it when you have some time @wfondrie?

Reply:
Ah yes, I should improve the doc string there. When used I think the models, the sequence is already a list. I should add that here too.

Reply:
Ok, good if that's the performance as intended. I fixed it on my side then.